source/samples/BLETest.cpp

```diff
@@ -71,6 +71,25 @@ void ble_test()
     // SECURITY_MODE_ENCRYPTION_NO_MITM, enables pairing without a passcode
     // SECURITY_MODE_ENCRYPTION_OPEN_LINK, pairing is no required
     //
+
+    uBit.messageBus.listen(MICROBIT_ID_BLE, MICROBIT_BLE_EVT_CONNECTED, onConnected);
+    uBit.messageBus.listen(MICROBIT_ID_BLE, MICROBIT_BLE_EVT_DISCONNECTED, onDisconnected);
+    
+    uBit.messageBus.listen(MICROBIT_ID_BLE_UART, MICROBIT_UART_S_EVT_DELIM_MATCH, onDelim);
+
+    new MicroBitAccelerometerService(*uBit.ble, uBit.accelerometer);
+    new MicroBitButtonService(*uBit.ble);
+    new MicroBitIOPinService(*uBit.ble, uBit.io);
+    new MicroBitLEDService(*uBit.ble, uBit.display);
+    new MicroBitMagnetometerService(*uBit.ble, uBit.compass);
+    new MicroBitTemperatureService(*uBit.ble, uBit.thermometer);
+
+    uart = new MicroBitUARTService(*uBit.ble, 32, 32);
+    uart->eventOn("\r\n");
+
+    if ( !uBit.compass.isCalibrated())
+        uBit.compass.calibrate();
```

Comment:
Let's calibrate after the BLE string is scrolled on the display to indicate to the user it's running the right thing, rather than jump straight into calibration.

Reply:
@JohnVidler the calibration has been moved to be performed after the `BLE ABDILMTU/P` text is scrolled, so this should be ready to be merged.